src/heif.imageio/heifinput.cpp

```diff
@@ -74,22 +75,20 @@ OIIO_EXPORT const char* heif_input_extensions[] = { "heic", "heif", "heics",
 OIIO_PLUGIN_EXPORTS_END
 
 
-#if 0
 bool
 HeifInput::valid_file(const std::string& filename) const
 {
-    const size_t magic_size = 12;
-    uint8_t magic[magic_size];
-    FILE *file = Filesystem::fopen(filename, "rb");
-    fread (magic, magic_size, 1, file);
-    fclose (file);
-    heif_filetype_result filetype_check = heif_check_filetype(magic,12);
+    FILE* fd = Filesystem::fopen(filename, "rb");
+    if (!fd)
+        return false;
+    uint8_t magic[12];
+    if (fread(magic, sizeof(magic), 1, fd) != sizeof(magic))
+        return false;
+    fclose(fd);
```

Comment:
This is more compactly expressed as

    uint8_t magic[12];
    if (Filesystem::read_bytes(filename, magic, sizeof(magic)) != sizeof(magic))
        return false;

Reply:
Looks good - changed